### PR TITLE
List spending txs

### PR DESCRIPTION
**Pull request by andrei-21**

### CODE REVIEW

File: src/wallet.rs [hunk 1]
```diff
@@ -412,6 +381,74 @@ impl Wallet {
...
+        wallet: &BdkWallet,
+        tip_height: u32,
+    ) -> LipaResult<TxDetails> {
+        let raw_tx = tx
+            .transaction
+            .as_ref()
+            .ok_or_else(|| permanent_failure("Tx does not have raw tx"))?;
+
+        let foreign_output = Self::find_foreign_output(&raw_tx.output, wallet)?
+            .ok_or_else(|| permanent_failure("None of tx outputs are foreign"))?;
```
Comment (danielgranhao) [inline src/wallet.rs:396]: I might be understanding this wrong, but it seems that this makes it such that if the wallet makes a tx to itself, the `get_spending_txs()` method will always fail. An error will be added to the `txs_details` iterator and the `try_collect` method will fail completely even if a single item in the iterator is an Error.

File: src/wallet.rs [hunk 2]
```diff
@@ -412,6 +381,74 @@ impl Wallet {
...
+
+    fn map_to_tx_details(
+        tx: TransactionDetails,
+        wallet: &BdkWallet,
+        tip_height: u32,
+    ) -> LipaResult<TxDetails> {
+        let raw_tx = tx
+            .transaction
+            .as_ref()
+            .ok_or_else(|| permanent_failure("Tx does not have raw tx"))?;
```
Comment (gcomte) [inline src/wallet.rs:393]: Didn't you introduce `ok_or_permanent_failure` for this case?

File: src/wallet.rs [hunk 3]
```diff
@@ -412,6 +381,74 @@ impl Wallet {
...
+        Ok(TxDetails {
+            id: tx.txid.to_string(),
+            output_address,
+            output_sat,
+            on_chain_fee_sat,
+            status: Self::to_tx_status(Some(tx), tip_height),
+        })
+    }
+
+    fn find_foreign_output(outputs: &Vec<TxOut>, wallet: &BdkWallet) -> LipaResult<Option<Script>> {
```
Comment (gcomte) [inline src/wallet.rs:421]: This function only returns the first instance of all foreign outputs (if there is any). This is not clear from the function name. Maybe needs clarification? Or would we want to have a `contains_foreign_output() -> bool` function?

File: src/wallet.rs (hunk 1 above)
Comment (danielgranhao) [reply src/wallet.rs:396]: I don't think so. If I understand correctly, the value `tx.sent` [is the sum of all owned inputs](https://docs.rs/bdk/0.26.0/bdk/struct.TransactionDetails.html#structfield.sent), meaning that a tx to the wallet itself will have `tx.sent > 0`. 

Maybe we drain to ourselves in the local wallet and have this scenario tested. Not that it's a thing that a user is likely to do, but if they do it by mistake the tx list could break.

**Review by danielgranhao — APPROVED** [review]

**Pull request merged**